workloads/files/workload-network-script-cm.yml

```diff
@@ -0,0 +1,500 @@
+      do
+        retries=0
+        while [ $retries -le 60 ] ; do
+          client_ready=`oc get replicationcontroller/uperf-client-${rc_num} -o json -n network0 | jq '.status.replicas==.status.availableReplicas'`
+          if [ "${client_ready}" == "true" ]; then
+            log "uperf-client-${rc_num} is ready"
+            break
+          else
+            log "Waiting for uperf-client-${rc_num} to be ready"
+            sleep 2
+          fi
+          retries=$[$retries + 1]
+        done
+        if [ "${client_ready}" == "false" ]; then
+          log "uperf-client-${rc_num} failed to be ready on time"
+          exit 1
+        fi
+        retries=0
+        while [ $retries -le 60 ] ; do
+          server_ready=`oc get replicationcontroller/uperf-server-${rc_num} -o json -n network0 | jq '.status.replicas==.status.availableReplicas'`
+          if [ "${server_ready}" == "true" ]; then
+            log "uperf-server-${rc_num} is ready"
+            break
+          else
+            log "Waiting for uperf-server-${rc_num} to be ready"
+            sleep 2
+          fi
+          retries=$[$retries + 1]
+        done
+        if [ "${server_ready}" == "false" ]; then
+          log "uperf-server-${rc_num} failed to be ready on time"
+          exit 1
+        fi
+      done
+
+      if [ "${NETWORK_TEST_SERVICE}" == "true" ]; then
+        clients=`oc get svc --output=json -n network0 --selector="name=client" | jq -r '[.items[].spec.clusterIP] | join(",")'`
+        servers=`oc get svc --output=json -n network0 --selector="name=server" | jq -r '[.items[].spec.clusterIP] | join(",")'`
+      else
+        clients=`oc get pods --output=json -n network0 --selector="name=client" | jq -r '[.items[].status.podIP] | join(",")'`
+        servers=`oc get pods --output=json -n network0 --selector="name=server" | jq -r '[.items[].status.podIP] | join(",")'`
+      fi
+
+      mv ~/.ssh/config ~/.ssh/config.bak || true
+      for host in $(echo "${clients} ${servers}" | sed "s/,/ /g")
+      do
+        HOST=${host} envsubst < /root/workload/ssh_config.template >> ~/.ssh/config
+      done
+      cat ~/.ssh/config.bak >> ~/.ssh/config || true
+      chmod 0600 ~/.ssh/config
+
+      pbench-uperf --test-types=${NETWORK_TEST_TYPE} \
+        --runtime=${NETWORK_TEST_RUNTIME} \
+        --message-sizes=${NETWORK_TEST_MESSAGE_SIZES} \
+        --protocols=${NETWORK_TEST_PROTOCOL} \
+        --instances=${NETWORK_TEST_INSTANCES} \
+        --samples=${NETWORK_TEST_SAMPLES} \
+        --max-stddev=10 \
+        --clients=${clients} \
+        --servers=${servers} \
+        --config=${config_prefix}_${pair_count}p_${NETWORK_TEST_INSTANCES//,/_}i_${NETWORK_TEST_PROTOCOL/,/_}
+
+      # Clean up project
+      if [[ "${NETWORK_TEST_CLEANUP}" == "true" ]]; then
+        oc delete project network0
+        retries=0
+        while [ $retries -le 60 ] ; do
+          remaining_projects=`oc get projects | egrep "^network[0-9]+ " -c || true`
```

Comment:
`--no-headers` would make things cleaner, but this is a nit.

Reply:
NP thanks!